h/views/api_auth.py

```diff
@@ -31,6 +32,43 @@ def __init__(self, context, request):
     @view_config(request_method='GET',
                  renderer='h:templates/oauth/authorize.html.jinja2')
     def get(self):
+        return self._authorize()
+
+    @view_config(request_method='GET',
+                 request_param='response_mode=web_message',
+                 renderer='h:templates/oauth/authorize.html.jinja2')
+    def get_web_message(self):
+        response = self._authorize()
+
+        if isinstance(response, HTTPFound):
```

Comment:
To double-check my understanding: would I be correct in thinking that this depends on the fact that `_authorize` *returns* an `HTTPFound` in the case of a successful authorization, but *raises* one when the user isn’t logged in?